pkgs/development/libraries/qt-6/hooks/qmake-hook.sh

```diff
@@ -0,0 +1,46 @@
+. @fix_qmake_libtool@
+
+qmakeFlags=( ${qmakeFlags-} )
+
+qmakePrePhase() {
+    qmakeFlags_orig=( "${qmakeFlags[@]}" )
+
+    # These flags must be added _before_ the flags specified in the derivation.
+    # "NIX_OUTPUT_DOC=${!outputDev}/${qtDocPrefix:?}" \
+    qmakeFlags=( \
+        "PREFIX=$out" \
+        "NIX_OUTPUT_OUT=$out" \
+        "NIX_OUTPUT_DEV=${!outputDev}" \
+        "NIX_OUTPUT_BIN=${!outputBin}" \
+        "NIX_OUTPUT_QML=${!outputBin}/${qtQmlPrefix:?}" \
+        "NIX_OUTPUT_PLUGIN=${!outputBin}/${qtPluginPrefix:?}" \
```

Comment:
Does these require an accompanying patch to really work?

Reply:
Yeah I messed up. Can either remove these since they aren't getting used, or put in more work and see if I can update that patch for 6.

Reply:
@NickCao any preference on that?

Reply:
Just add a TODO?

Reply:
Done